### PR TITLE
Task-36001: Allow login with username or email

### PR DESCRIPTION
**Pull request by azayati**

### CODE REVIEW

**Comment by rdenarie** [comment]
@azayati Did you check if it works with and LDAP or AD ?

**Review by rdenarie — DISMISSED** [review]

**Comment by rdenarie** [comment]
Did you remove the webui properties because it will be treaten by crowdin ?

File: component/web/security/src/main/java/org/exoplatform/web/login/LoginServlet.java [hunk 1]
```diff
@@ -171,6 +176,24 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         int status;
         if (req.getRemoteUser() == null) {
             if (username != null && password != null) {
+                // email authentication
+                if (username.contains("@")) {
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/LoginServlet.java:180]: and what if username contains an @ ?

Comment (azayati) [reply component/web/security/src/main/java/org/exoplatform/web/login/LoginServlet.java:180]: basically, the username with @ is forbidden, but even if the username contains @ (with dirty creation), we will not find a user with email == username (with the @) and the login will then be executed normally

**Review by rdenarie — APPROVED** [review]

**Pull request merged**